workers/unity/Assets/Playground/Scripts/EntityInitialization/EntityGameObjectCreator.cs

```diff
@@ -1,53 +1,82 @@
-using System;
 using System.Collections.Generic;
+using System.IO;
+using Generated.Improbable;
 using Improbable.Gdk.Core;
-using Improbable.Worker;
-using Unity.Entities;
+using Improbable.Gdk.Core.GameObjectRepresentation;
 using UnityEngine;
-using Object = UnityEngine.Object;
+using Transform = Generated.Improbable.Transform.Transform;
 
 namespace Playground
 {
-    public class EntityGameObjectCreator
+    public class EntityGameObjectCreator : IEntityGameObjectCreator
     {
-        private readonly World world;
-        private readonly Dictionary<string, GameObject> cachedPrefabs;
+        private readonly Dictionary<string, GameObject> cachedPrefabs
+            = new Dictionary<string, GameObject>();
 
-        public EntityGameObjectCreator(World world, Dictionary<string, GameObject> initialCachedPrefabs = null)
+        private readonly string workerType;
+
+        public EntityGameObjectCreator(string workerType)
         {
-            this.world = world;
-            cachedPrefabs = initialCachedPrefabs ?? new Dictionary<string, GameObject>();
+            this.workerType = workerType;
         }
 
-        public GameObject CreateEntityGameObject(Entity entity, string prefabPath, Vector3 position,
-            Quaternion rotation, EntityId spatialEntityId)
+        public GameObject GetGameObjectForEntityAdded(SpatialOSEntity entity, WorkerSystem worker)
```

Comment:
This doesn't "get" a gameobject, it creates one

Reply:
It might create one. It doesn't have to. It might have already been made.

Reply:
This doesn't support that anyways. The prefab is either in the worker specific folder or the common folder.

Reply:
It doesn't have to create, but I'll still rename it back to create because that's the simplest way to use it and that's what the name of the interface is. Power users can just use it with stuff other than straightforward instantiation anyway